code/modules/surgery/organs/external/restyling.dm

```diff
@@ -0,0 +1,89 @@
+
+	valid_restyles = list()
+	for(var/datum/sprite_accessory/accessory as anything in get_global_feature_list())
+		if(initial(accessory.locked)) //locked is for stuff that shouldn't appear here
+			continue
+		valid_restyles.Add(accessory)
+
+	return valid_restyles
+
+///Invokes async to ask a mob if their limb if their limb has an external organ
+/mob/living/carbon/proc/_attempt_feature_restyle(atom/source, mob/living/trimmer, atom/movable/original_target, body_zone, restyle_type, style_speed)
+	SIGNAL_HANDLER
+
+	INVOKE_ASYNC(src, PROC_REF(attempt_feature_restyle), source, trimmer, original_target, body_zone, restyle_type, style_speed)
+
+///Asks the mob to ask their limb about restyling
+/mob/living/carbon/proc/attempt_feature_restyle(atom/source, mob/living/trimmer, atom/movable/original_target, body_zone, restyle_type, style_speed)
+	var/obj/item/bodypart/targeted_part = get_bodypart(body_zone)
+	if(!targeted_part)
+		to_chat(trimmer, span_warning("There's no bodypart there!"))
+		return FALSE
+	targeted_part.attempt_feature_restyle(source, trimmer, original_target, body_zone, restyle_type, style_speed)
+	return TRUE
+
+///Invoke async so we dont break signals
+/obj/item/bodypart/proc/_attempt_feature_restyle(atom/source, mob/living/trimmer, atom/movable/original_target, body_zone, restyle_type, style_speed)
+	SIGNAL_HANDLER
+
+	INVOKE_ASYNC(src, PROC_REF(attempt_feature_restyle), source, trimmer, original_target, body_zone, restyle_type, style_speed)
+
+///Asks the external organs inside the limb if they can restyle
+/obj/item/bodypart/proc/attempt_feature_restyle(atom/source, mob/living/trimmer, atom/movable/original_target, body_zone, restyle_type, style_speed)
+	var/list/valid_features = list()
+	for(var/obj/item/organ/external/feature as anything in external_organs)
+		if(feature.restyle_flags & restyle_type)
+			valid_features.Add(feature)
+
+	var/obj/item/organ/external/target_organ
+	switch(LAZYLEN(valid_features))
+		if(1)
+			target_organ = valid_features[1]
+		if(2 to INFINITY)
+			var/choose_options = list()
+			var/name_to_organ = list() //literally so I dont have to loop again after someones made their choice
+			for(var/obj/item/organ/external/organ_choice as anything in valid_features)
+				choose_options[organ_choice.name] = image(organ_choice)
+				name_to_organ[organ_choice.name] = organ_choice
+			var/picked_option = show_radial_menu(trimmer, original_target, choose_options, radius = 38, require_near = TRUE)
+			if(picked_option)
+				target_organ = name_to_organ[picked_option]
+			else
+				return
+		else
+			to_chat(trimmer, span_warning("There are no restylable features there!"))
+			return
+
+	target_organ.attempt_feature_restyle(source, trimmer, original_target, body_zone, restyle_type, style_speed)
+
+///Invoke async so we dont break signals
+/obj/item/organ/external/proc/_attempt_feature_restyle(atom/source, mob/living/trimmer, atom/movable/original_target, body_zone, restyle_type, style_speed)
```

Comment:
This should be something like `on_attempt_feature_restyle` instead IMO

Reply:
Sure!